rascal-vscode-extension/src/RascalExtension.ts

```diff
@@ -148,6 +147,32 @@ export class RascalExtension implements vscode.Disposable {
         }
     }
 
+    private getTerminalOrigin(uri: vscode.Uri | undefined): string {
+        if (uri) {
+            const config = vscode.workspace.getConfiguration();
+            const originFormat = config.get('rascal.terminal.name.originFormat');
+            switch (originFormat) {
+                case 'Project root': {
+                    const projectRoot = vscode.workspace.getWorkspaceFolder(uri);
+                    if (projectRoot && projectRoot.name) {
+                        return projectRoot.name;
+                    }
+                    break;
+                }
+                case 'Module': {
+                    const sep = uri.path.lastIndexOf("/");
```

Comment:
I would prefer to show the fully qualitfied name of the module, not just the last part. I think we have this calculated already?